File: 1072-flip-columns-for-maximum-number-of-equal-rows/1072-flip-columns-for-maximum-number-of-equal-rows.py
class Solution:
    def maxEqualRowsAfterFlips(self, matrix):
        m = len(matrix[0])
        max_count = 0
        freq = defaultdict(int)
        
        for row in matrix:
            original = "".join(str(cell) for cell in row)
            flipped = "".join(str(1 - cell) for cell in row)
            freq[original] += 1
            freq[flipped] += 1
            max_count = max(max_count, freq[original], freq[flipped])
        
        return max_count

File: 1072-flip-columns-for-maximum-number-of-equal-rows/1072-flip-columns-for-maximum-number-of-equal-rows.java
class Solution {
    public int maxEqualRowsAfterFlips(int[][] matrix) {
        int m = matrix[0].length;
        int maxCount = 0;
        HashMap<String, Integer> map = new HashMap<>();
        
        for (int[] row : matrix) {
            StringBuilder original = new StringBuilder();
            StringBuilder flipped = new StringBuilder();
            for (int cell : row) {
                original.append(cell);
                flipped.append(1 - cell);
            }
            String orgStr = original.toString();
            String flipStr = flipped.toString();
            
            map.put(orgStr, map.getOrDefault(orgStr, 0) + 1);
            map.put(flipStr, map.getOrDefault(flipStr, 0) + 1);
            
            maxCount = Math.max(maxCount, map.get(orgStr));
            maxCount = Math.max(maxCount, map.get(flipStr));
        }
        
        return maxCount;
    }
}
